File: src/main/scala/s/decorator/delegate/SideBorderDelegate.scala
package s.decorator.delegate

final class SideBorderDelegate(private val display: Display, private val borderChar: Char) extends DisplayDelegate {
  override def columns: Int = 1 + display.columns + 1

  override def rows: Int = display.rows

  override def rowText(row: Int): String = borderChar.toString + display.rowText(row) + borderChar.toString
}

File: src/main/java/j/decorator/delegate/SideBorderDelegate.java
package j.decorator.delegate;

public final class SideBorderDelegate implements DisplayDelegate {
  private final Display display;
  private final char borderChar;

  public SideBorderDelegate(Display display, char ch) {
    this.display = display;
    borderChar = ch;
  }

  @Override
  public int getColumns() {
    return 1 + display.getColumns() + 1;
  }

  @Override
  public int getRows() {
    return display.getRows();
  }

  @Override
  public String getRowText(int row) {
    return borderChar + display.getRowText(row) + borderChar;
  }
}
